lib/tree.rb

```diff
@@ -907,5 +907,35 @@ def print_tree(level = 0, max_depth = nil, block = lambda { |node, prefix|  puts
       children { |child| child.print_tree(level + 1, max_depth, block) if child } # Child might be 'nil'
     end
 
+        # Node path.
+    #
+    # If a block is provided, yields each of the sibling nodes to the block.
+    # The root always has +nil+ siblings.
+    #
+    # @param [String] separator The separator between the nodes.
+    #
+    # @return [String] The path from the root to the node in string format
+    def path_as_string(separator)
+      get_path_array.reverse.join(separator)
+    end
+
+    # Node path.
+    #
+    # If a block is provided, yields each of the sibling nodes to the block.
+    # The root always has +nil+ siblings.
+    #
+    # @return [Array<String>] An array with the nodes from the root to the node itself
+    def path_as_array
+      get_path_array.reverse
+    end
+
+    def get_path_array(current_array_path = [])
+      path_array = current_array_path + [name]
```

Comment:
If you like we could make also something like:

```
...
  def get_path_array(current_array_path = [], attribute=:name)
     path_array = current_array_path + [send(attribute)]
...
```

But I don't know if it's worth to do so. 
In my case the Name is and Id and the Content is a Name, so it made sense to have a dynamic way to select the attribute.